llvm/lib/Target/RISCV/RISCVInstrInfoCOREV.td

```diff
@@ -974,14 +974,14 @@ let Predicates = [HasExtXcvmem] in {
             (CV_LHU_rr GPR:$base, GPR:$offset)>;
   def : Pat<(extloadi16 (AddrRegReg GPR:$base, GPR:$offset)),
             (CV_LHU_rr GPR:$base, GPR:$offset)>;
-  def : Pat<(load (AddrRegReg GPR:$base, GPR:$offset)),
+  def : Pat<(i32 (load (AddrRegReg GPR:$base, GPR:$offset))),
             (CV_LW_rr GPR:$base, GPR:$offset)>;
 
   def : Pat<(post_truncsti8 GPR:$val, GPR:$base, simm12:$offset),
             (CV_SB_ri_inc GPR:$val, GPR:$base, simm12:$offset)>;
   def : Pat<(post_truncsti16 GPR:$val, GPR:$base, simm12:$offset),
             (CV_SH_ri_inc GPR:$val, GPR:$base, simm12:$offset)>;
-  def : Pat<(post_store GPR:$val, GPR:$base, simm12:$offset),
+  def : Pat<(post_store (i32 GPR:$val), GPR:$base, simm12:$offset),
```

Comment:
Doesn't the pattern at line 991 need this fix too? Just asking in case I'm missing something

Reply:
Apologies for the mistake! I have fixed it in the latest commit.